auth/auth/auth.py

```diff
@@ -192,6 +192,16 @@ def _delete(key):
     _delete('flow')
 
 
+def validate_next_page_url(next_page):
+    if not next_page:
+        raise web.HTTPBadRequest('Invalid next page: empty')
+    valid_next_services = ['batch', 'auth']
+    for service in valid_next_services:
+        if next_page.startswith(deploy_config.external_url(service, '')):
```

Comment:
This is probably done best with a URL parser. You can start a URL with another domain, like those scam emails that are supposedly from the post office, but are instead usps.com-oops.net or something similar.